packages/next/client/portal/index.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react'
+import { createPortal } from 'react-dom'
+
+type PortalProps = {
+  children: React.ReactNode
+  type: string
+}
+
+export const Portal: React.FC<PortalProps> = ({ children, type }) => {
+  let [isMounted, setMounted] = React.useState(false)
+  let mountNode = React.useRef<HTMLDivElement | null>(null)
+  let portalNode = React.useRef<HTMLElement | null>(null)
+  let [, forceUpdate] = React.useState<{}>()
+  React.useEffect(() => {
+    setMounted(true)
+  }, [])
```

Comment:
Why is this effect needed, given we already check `mountNode.current` in the second effect? Or do you want `<Portal>` to output `null` when pre-rendering and hydrating intentionally?

Reply:
The intention was to keep it out of the initial markup from the server. This may not be a big deal, however. Play with it if you'd like to code-golf the bytes away!

If it is present from the server, we should probably mark it with an attribute so the dev knows what it is, instead of wondering where a random `<span/>` came from.